tests/ga4gh/broker/endpoints/test_builds.py

```diff
@@ -0,0 +1,412 @@
+            assert 'id' in res and res['id'][:6] == MOCK_REPOSITORIES[1]['id']
+
+    @patch('broker.ga4gh.broker.endpoints.builds.create_build',
+           mocked_create_build)
+    def test_register_builds_duplicate_key_error(self):
+        self.setup()
+        mock_resp = MagicMock(side_effect=DuplicateKeyError(''))
+        self.app.config['FOCA'].db.dbs['brokerStore'].collections['builds']. \
+            client.insert_one = mock_resp
+        with self.app.app_context():
+            with pytest.raises(InternalServerError):
+                register_builds(MOCK_REPOSITORIES[1]['id'],
+                                MOCK_REPOSITORIES[1]['access_token'],
+                                MOCK_BUILD_PAYLOAD)
+
+    @patch('broker.ga4gh.broker.endpoints.builds.create_build',
+           mocked_create_build)
+    def test_register_builds_unauthorized(self):
+        self.setup()
+        with self.app.app_context():
+            with pytest.raises(Unauthorized):
+                register_builds(MOCK_REPOSITORIES[1]['id'],
+                                'access_token',
+                                MOCK_BUILD_PAYLOAD)
+
+    @patch('broker.ga4gh.broker.endpoints.builds.create_build',
+           mocked_create_build)
+    def test_register_builds_repository_not_found(self):
+        self.setup()
+        with self.app.app_context():
+            with pytest.raises(RepositoryNotFound):
+                register_builds('id',
+                                MOCK_REPOSITORIES[1]['access_token'],
+                                MOCK_BUILD_PAYLOAD)
+
+    def test_get_builds(self):
+        self.setup_with_build()
+        with self.app.app_context():
+            res = get_builds(MOCK_REPOSITORIES[1]['id'])
+            assert isinstance(res, list)
+            assert len(res) == 2
+
+    def test_get_builds_build_not_found(self):
+        self.setup()
+        with self.app.app_context():
+            with pytest.raises(BuildNotFound):
+                get_builds(MOCK_REPOSITORIES[1]['id'])
+
+    def test_get_builds_repository_not_found(self):
+        self.setup()
+        with self.app.app_context():
+            with pytest.raises(RepositoryNotFound):
+                get_builds('abcd')
+
+    def test_get_build_info(self):
+        self.setup_with_build()
+        with self.app.app_context():
+            res = get_build_info(MOCK_BUILD_INFO['id'])
+            assert isinstance(res, dict)
+            assert 'finished_at' in res
+            assert 'head_commit' in res
+            assert 'started_at' in res
+            assert 'branch' in res['head_commit']
+
+    def test_get_build_info_build_not_found(self):
+        self.setup_with_build()
+        with self.app.app_context():
+            with pytest.raises(BuildNotFound):
+                get_build_info('abcd')
+
+    def test_git_clone_and_checkout(self):
+        clone_path = git_clone_and_checkout(
+            repo_url="https://github.com/akash2237778/Broker-test",
```

Comment:
Please search and replaces all occurrences of this and other URLs. You may use a common global variable.

Also the repository Broker-test should be copied to elixir-cloud-aai as well